packages/db/postgres/functions.sql

```diff
@@ -251,17 +251,38 @@ BEGIN
 END
 $$;
 
-CREATE OR REPLACE FUNCTION user_used_storage(query_user_id BIGINT) RETURNS TEXT
-  LANGUAGE plpgsql
+CREATE TYPE used_storage AS (uploaded TEXT, pinned TEXT);
+
+CREATE OR REPLACE FUNCTION user_used_storage(query_user_id BIGINT) 
+RETURNS used_storage
+LANGUAGE plpgsql
 AS
 $$
+DECLARE used_storage used_storage;
 BEGIN
-  return(
-    select sum(c.dag_size)
-    from upload u
-    join content c on c.cid = u.content_cid
-    where u.user_id = query_user_id and u.deleted_at is null
-  )::TEXT;
+  SELECT COALESCE(SUM(c.dag_size), 0)
+  INTO used_storage.uploaded::TEXT
+  FROM upload u
+  JOIN content c ON c.cid = u.content_cid
+  WHERE u.user_id = query_user_id::BIGINT
+  AND u.deleted_at is null;
+
+  -- NOTE: auth_key.deleted_at is NOT included in the where clause, since the 
+  -- auth_key is only used to authorize a request.
+  SELECT COALESCE((
+    SELECT COALESCE(SUM(c.dag_size), 0)
+    FROM psa_pin_request psa_pr
+    JOIN content c ON c.cid = psa_pr.content_cid
+    JOIN pin p ON p.content_cid = psa_pr.content_cid
+    JOIN auth_key a ON a.id = psa_pr.auth_key_id
```

Comment:
This query is dangerous in terms of execution time needed considering we are using 4 tables.

We should add an index for `psa_pr.content_cid`. Also not entirely sure if we need an individual index for `p.content_cid`. Can you do some benchmarks on these?

Reply:
So, I think we did not get super better results in `psa_pin_request.content_cid` index because we still have a low number of pin requests. 

@alanshaw @hugomrdias what do you think about these indexes?

Reply:
@vasco-santos I'm happy to get this PR shipped without the indexes and monitor it. We can follow up with a PR to add indexes on all the foreign key source columns that are used in joins.